### ISSUE TITLE
Add ability to set different tags for wrapper vs link

### ISSUE THREAD

**Issue by kidbombay** [tuvistavie/phoenix-active-link#10]
Bootstrap expects active links like this
```
<li class="nav-item active">
  <a class="nav-link" href="/about">About</a>
</li>
```

But when you set the active tag like this
```
<%= active_link(@conn, "About", to: page_path(@conn, :about), wrap_tag: :li, class_active: "active nav-item", class_inactive: "nav-item") %>
```
The output isn't correct. I want apply nav-link to the inner link and no other classes.
```
<li class="active nav-item">
  <a class="active nav-item" href="/about">About</a>
</li>
```

**Comment by taorg** [comment]
+1 
Hello @kidbombay yea, I found the same issue working with Bootstrap, but more than tags is a matter of CSS classes, what I think its needed is the ability to set different active/inactive classes for tag and wrapper.
Anyway, yes, it would be nice to have it.

**Comment by sn3p** [comment]
+1 would be nice to have control over the wrapper and link classes.

@danhper thoughts?

**Comment by danhper** [comment]
I agree this would be a good feature to have. If you would like to send a PR, I am happy to merge it.
Thanks!

**Comment by sn3p** [comment]
I think it's worth mentioning the issues with Bootstrap described above can be resolved using the `class` attribute:

```eex
<%= active_link(@conn, "About",
      to: "/about",
      wrap_tag: :li,
      class: "nav-link",
      class_active: "active nav-item",
      class_inactive: "nav-item",
      active: true) %>
```

```html
<li class="active nav-item">
  <a class="active nav-item nav-link" href="/about">About</a>
</li>
```

Might be a good idea to document this.

**Comment by sn3p** [comment]
> I agree this would be a good feature to have. If you would like to send a PR, I am happy to merge it.

Sure! I guess this would involve adding/changing options. Any suggestions?

First of all I think instead of `class_inactive` we could maybe use `class` and append `class_active` when active? I'm aware this is a breaking change, so it's up to you :)

```ex
active_link(@conn, "About", to: "/about", class: "nav-item", class_active: "active")
```

In addition we could introduce new `wrap_*` options like:

```eex
<%= active_link(@conn, "About",
      to: "/about",
      class: "nav-link",
      class_active: nil,
      wrap_tag: :li,
      wrap_class: "nav-item",
      wrap_class_active: "active") %>
```

```html
<li class="nav-item active">
  <a class="nav-link" href="/about">About</a>
</li>
```

Alternatively we could keep the `class_inactive` behaviour and introduce `wrap_class_active` and `wrap_class_inactive` options:

```eex
<%= active_link(@conn, "About",
      to: "/about",
      class_inactive: "nav-link",
      class_active: "nav-link",
      wrap_tag: :li,
      wrap_class_inactive: "nav-item",
      wrap_class_active: "nav-item active") %>
```

```html
<li class="nav-item active">
  <a class="nav-link" href="/about">About</a>
</li>
```

**Comment by danhper** [comment]
Hi @sn3p ,
Thanks a lot for the suggestions.
I personally like the `wrap_*` approach. I think it makes things clear and flexible.
If you would have time to send a PR it would be wonderful.

**Comment by sn3p** [comment]
Will give this a shot 👍 

How do you feel about deprecating `class_inactive` in favour of `class`?
And use `wrap_class` instead of `wrap_class_inactive`?

**Comment by danhper** [comment]
Thanks a lot.
I do not have a strong opinion on this but I think they are two different things.
`class` should be present all the time. `class_inactive` should not be present if the element is active.
I am very unsure there is a use case for `class_inactive` so I am happy to go with only `class` and deprecate `class_inactive`.

**Comment by lucasprag** [comment]
Just my two cents here. I find the `class_inactive` particulary useful when using more functional CSS frameworks like tailwindcss. See my example:

```eex
<%= active_link(
  @conn, "Blogs",
  to: Routes.blog_path(@conn, :index),
  active: [{TrailReaderWeb.BlogController, :index}],
  class: "px-3 py-2 rounded-md text-sm",
  class_inactive: "font-bold",
  class_active: "bg-primary text-navbar font-medium") %>
```

Notice that the `font-bold` class is only applied for inactive links and `font-medium` only for active ones.

If I were to rely only on `class` and `class_active`, I wouldn't be able to use tailwind by itself because the font would always be bold.

```eex
<%= active_link(
  @conn, "Blogs",
  to: Routes.blog_path(@conn, :index),
  active: [{TrailReaderWeb.BlogController, :index}],
  class: "px-3 py-2 rounded-md text-sm font-bold",
  class_active: "bg-primary text-navbar font-medium") %>
```

So please don't remove this option. Thanks.
Unless I'm missing something?
